Updates of the PI18 Protocol for the Mppsolar LV5048 Hyprid V2 inverter.

preussal opened the pull request:
After some training time and a lot of try and error.
C ++ is not mine.

With the update, except for the battery Bulk/Float Voltage, and battery Charge/Discharge Voltage work all important things.

The Selects can be adjusted via the YML. But these values are the one that supports the protocoll. Sensor and Device classes were adjusted so that the values in the home assistant are also suitable.

Webinterface
![grafik](https://github.com/syssi/esphome-pipsolar/assets/26431172/064f58d3-e053-4d89-88d7-6c685d5d619d)


If someone has an idea of how to do two Select fields with the values for the battery voltage Bulk/Float, that would be great
Or a text field where you then enter "552, 540". Which is then translated into the command "^S015MCHGV552, 540".

For the current mapping solution, 2x 11025 lines would be required to enable all values for it.

Commit:
Rename P007GS to P005GS for MppSolar LV5048 Hybrid V2

Commit:
Try output_source_priority same as PIP8048

Commit:
add Select

Commit:
add Class

Commit:
add Select

Commit:
Select

Commit:
Update

Commit:
test

Commit:
Revert "test"

This reverts commit 17042f637bccc173b32a5b8af54002ac6744dc15.

Commit:
output_source_priority as Select

Commit:
Add charger source priority Select

Commit:
Add charger source priority Select

Commit:
Add solar power priority Select

Commit:
Add solar power priority Select

Commit:
Add solar power priority Select

Commit:
Grummel

Commit:
Grummel

Commit:
Grummel

Commit:
Grummel

Commit:
Add Machine Type Selector Grid-Tie / Off-Grid

Commit:
Add Machine Type Selector Grid-Tie / Off-Grid

Commit:
Switch options from Grid-Tie OFF-Grid

Commit:
update

Commit:
add input voltage range Selector Appliances or UPS

Commit:
Cleanup

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
try battery_power_direction_

Commit:
Revert "Cleanup"

This reverts commit da0d48afeb053e4715dbd4e6510b06fb4586e758.

Commit:
Reset to cleanup

Commit:
Merge branch 'pi18_lv5048-Hybrid-V2' of github.com:preussal/esphome-pipsolar into pi18_lv5048-Hybrid-V2

Commit:
battery_power_direction

Commit:
battery_power_direction

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
battery_power_direction next try

Commit:
cleanup

Commit:
cleanup

Commit:
cleanup

Commit:
add current_max_charging_current and current_max_ac_charging_current

Commit:
cleanup

Commit:
cleanup

Commit:
load_connection as Text

Commit:
load_connection as Text

Commit:
load_connection as Text

Commit:
load_connection as Text

Commit:
Remove Power Saving LV5048 Hybrid V2 not have this option

Commit:
Remove Power Saving LV5048 Hybrid V2 not have this option

Commit:
Remove Power Saving LV5048 Hybrid V2 not have this option

Commit:
Remove Power Saving LV5048 Hybrid V2 not have this option

Commit:
Remove Power Saving LV5048 Hybrid V2 not have this option

Commit:
Remove Power Saving LV5048 Hybrid V2 not have this option

Commit:
Remove Power Saving LV5048 Hybrid V2 not have this option

Commit:
add mppt1_charger_status as Text

Commit:
add line_power_direction as Text

Commit:
Naming

Commit:
add dc_ac_power_direction as Text

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh

Commit:
Total generate Energie are in Watt Hours, now in kWh and the right device:class

Commit:
Total generate Energie are in Watt Hours, now in kWh and the right device:class

Commit:
right device class and state class

Commit:
battery right device class and state class

Commit:
battery type as text

Commit:
battery type as text

Commit:
Battery_type are Settabal but you do this normaly one Time

Commit:
Total generate Energie are in Watt Hours, now in kWh and the right device:class

Commit:
Total generate Energie are in Watt Hours, kwh convert not working

Commit:
esp32_example with working setting

Commit:
test output battery_under_voltage

Commit:
test output battery_under_voltage

Commit:
test output battery_under_voltage

Commit:
select output battery_under_voltage

Commit:
select output battery_under_voltage

Commit:
select output battery_under_voltage

Commit:
select output battery_under_voltage

Commit:
select output battery_under_voltage

Commit:
select output battery_under_voltage

Commit:
select output battery_under_voltage

Commit:
select output battery_under_voltage

Commit:
select output battery_under_voltage

Commit:
Cleanup + CONF_BATTERY_BULK_VOLTAGE options + CONF_BATTERY_FLOAT_VOLTAGE options

Commit:
CONF_BATTERY_BULK_VOLTAGE and CONF_BATTERY_FLOAT_VOLTAGE

Commit:
test

Commit:
test

Commit:
test

Commit:
test

Commit:
Updates of the PI18 Protocol for the Mppsolar LV5048 Hyprid V2 inverter.
With the update, except for the battery Bulk/Float Voltage, and battery Charge/Discharge Voltage work all important things.

The Selects can be adjusted via the YML. But these values are the one that supports the protocoll.
Sensor and Device classes were adjusted so that the values in the home assistant are also suitable.

Commit:
Update

Commit:
update

daniil226 commented:
Hello. How can I connect your version of the library? The 007GS command does not work in the original. The inverter only responds to 005GS, but when I connect your branch, it gives an error during compilation.

preussal commented:
Of course you can use my branch
Here is my external source.
Can you show me the compile error, you are also welcome to open an Issue in on my Dev Repo.

```
external_components:
  #- source: github://preussal/esphome-pipsolar@PI18_LV5048_Hybrid_V2
  - source: github://preussal/esphome-pipsolar@PI18_LV5048_Hybrid_V2_DEV
  #- source: github://syssi/esphome-pipsolar@pi18
    refresh: 0s
```

I just let the build go through, and I had no build mistakes.

preussal committed:
Merge pull request #1 from preussal/PI18_LV5048_Hybrid_V2_DEV

Pi18 lv5048 hybrid v2 dev

daniil226 commented:
I don’t know what you did, but now everything compiled without errors. Everything worked. Thank you very much. 
Inverter AXIOMA 5 kW

daniil226 commented:
errors that were described in the topic:
https://github.com/syssi/esphome-pipsolar/discussions/139

preussal commented:
> I don’t know what you did, but now everything compiled without errors. Everything worked. Thank you very much. Inverter AXIOMA 5 kW

That was my mistake, I always built my ESP on the DEV branch. 
For weeks / months. 
But also not further developed because what I needed works.

And the end, i forgot to Merge all the changes to the Stable Branch.
Sorry

preussal committed:
Update __init__.py

WARNING Using `select.SELECT_SCHEMA` is deprecated and will be removed in ESPHome 2025.11.0. Please use `select.select_schema(...)` instead.

preussal committed:
Update __init__.py

preussal committed:
Update __init__.py

preussal committed:
Update __init__.py

preussal committed:
rollback

preussal committed:
Update __init__.py

preussal committed:
Update __init__.py

Fix *.*_SCHEMA deprecations

preussal committed:
Update __init__.py

preussal committed:
Merge pull request #2 from preussal/PI18_LV5048_Hybrid_V2_DEV

Pi18 lv5048 hybrid v2 dev

syssi committed:
Resolve merge conflict with pi18 branch in sensor/__init__.py
